src/ngraph/frontend/onnx_import/utils/broadcasting.cpp

```diff
@@ -54,53 +55,102 @@ static std::vector<ngraph::Shape> calculate_numpy_broadcast_shape(ngraph::Shape
     return {output_shape, left_shape, right_shape};
 }
 
+/// \brief      Broadcast input node.
+///
+/// \note       The source shape does not have to be the actual shape of input node. However
+///             it should be a superset of it (containing it as a continuous subset). This implies
+///             we may expand the number of axes of input node.
+///
+/// \param[in]  node          The input Node to be broadcasted.
+/// \param[in]  output_shape  The output shape.
+/// \param[in]  source_shape  The source shape from which we want to broadcast input node.
+///
+/// \return     The boroadcasted Node.
+///
+static std::shared_ptr<ngraph::Node> broadcast(const std::shared_ptr<ngraph::Node>& node,
+                                               const ngraph::Shape& output_shape,
+                                               const ngraph::Shape& source_shape)
+{
+    ngraph::AxisVector broadcast_axes;
+    ngraph::Shape squeezed_shape;
+    // Positions of axes which have length of 1 are needed to calculate broadcast_axes
+    // for nGraph broadcast operation. We need to remove all ones from source shape
+    // to avoid broadcasting axis conflict.
+    for (auto index = 0; index < output_shape.size(); ++index)
+    {
+        (source_shape.at(index) == 1) ? broadcast_axes.push_back(index)
```

Comment:
This is the most unusual syntax I have seen in a while. Please use if/else instead. It is way easier to read.